tests/components/mqtt/test_trigger.py

```diff
@@ -213,8 +214,9 @@ async def test_if_not_fires_on_topic_but_no_payload_match(hass, calls):
     assert len(calls) == 0
 
 
-async def test_encoding_default(hass, calls, mqtt_mock):
+async def test_encoding_default(hass, calls, mqtt_mock_entry):
```

Comment:
There's an automatically used mock `setup_comp`, we should not need the `mqtt_mock_entry` here?